src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,432 @@
+package org.broadinstitute.hellbender.cmdline.GATKPlugin;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineException;
+import org.broadinstitute.barclay.argparser.CommandLinePluginDescriptor;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.walkers.annotator.Annotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.reflections.ReflectionUtils;
+
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A plugin descriptor for managing the dynamic discovery of both @{InfoFieldAnnotation} and @{GenotypeAnnotation} objects
+ * within the @{org.broadinstitute.hellbender.tools.walkers.annotator} package which handles integrating annotation specific
+ * arguments from the command line with tool specified defaults.
+ *
+ * Unlike @{GATKReadFilterPluginDescriptor} annotation order is not important and thus argument order is not guarinteed to be
```

Comment:
I think that annotation order might be important if the annotation uses information for previous annotations (e.g., DP can be computed with AD for a concrete annotation, without require anything else).

As an API user, I would love to have this plugin working for custom annotations and re-use the code in GATK with some tweaks (e.g., custom args). Please, take in consideration these use cases too.

Reply:
I agree - I don't see any reason that we can't or shouldn't preserve the order.

Reply:
Hmm, this raises an interesting question, what is the proper order when there are group annotations involved? Its unclear to me if those should come first or last. Right now they will simply be added first in the order that they are discovered by the Barclay reflection (Which I assume is alphabetical)